### PR TITLE
add ginkgo package, use it in dealii along with symengine

### PR DESCRIPTION
**Pull request by davydden**
@pratikvn @gflegar @tcojean  FYI.

~~blocked by~~ https://github.com/ginkgo-project/ginkgo/issues/270 EDIT: added some workarounds suggested upstream.

### CODE REVIEW

File: var/spack/repos/builtin/packages/dealii/package.py [hunk 1]
```diff
@@ -61,6 +62,8 @@ class Dealii(CMakePackage, CudaPackage):
             description='Compile with Sundials')
     variant('slepc',    default=True,
             description='Compile with Slepc (only with Petsc and MPI)')
+    variant('symengine', default=False,
+            description='Compile with Symengine')
```
Comment (jppelteret) [inline var/spack/repos/builtin/packages/dealii/package.py]: `Compile with Symengine` -> `Compile with SymEngine`

**Comment by tcojean** [comment]
Thanks for this @davydden ! We will take care of the issues you found ASAP.

**Comment by davydden** [comment]
@tcojean 

> We will take care of the issues you found ASAP.

thanks 👍

**Comment by davydden** [comment]
@alalazo @adamjstewart could you have a look?

**Pull request merged**